### DISCUSSION TITLE
Add before send email message event

### DISCUSSION THREAD

**Discussion by nateiler** [craftcms/cms#8523] [Ideas]
We have a scenario where we'd like to check a blacklist prior to sending to a recipient.  We have identified bounced email addresses and would like to suppress subsequent sends... and all we need is an event prior to send which will:

1) Not send if event is not valid.
2) Access to the Message object (in order to remove one of many recipients).

We could replace the entire component, but this might be useful for others a well.

**Comment by brandonkelly** [comment]
Care to submit a PR?

**Comment by nateiler** [comment]
What are you busy prepping for GA? 

Most certainly.
